lib/utils.js

```diff
@@ -115,4 +115,23 @@ utils.mergeDeep = (target, ...sources) => {
     return target;
 };
 
+utils.exportGlobal = (global) => {
+    if (!global || ['string', 'number', 'boolean'].includes(typeof global)) {
+        return global;
+    }
+    if (Array.isArray(global)) {
+        return global.map((x) => utils.exportGlobal(x));
+    }
+    if (global.constructor === Object) {
```

Comment:
What if a global is an instance of some class?

Reply:
We will just return it unmodified and rely on stable JSON stringification for figuring out if it's the same object.

Reply:
Thus, we pretty much filter out loggers and methods huh? 

Json-stable-stringify already drops out the 'function' properties:

```
console.log(stringify({
    bla: 'blabla',
    test: function (bla) {
        return 'blabla';
    }
}));

{"bla":"blabla"}
```

thus the condition on line 127 is not needed. 

So all we are left with it filtering out the logger right? Different loggers json-stricgified should be only able to differ in parameters, that are only modified by logger.child. Given that this is only executed on startup, where we don't create child loggers, this code should not be needed at all.

And if it's left in, we would basically be ignoring the different loggers we might wanna pass into different module initializers.

Reply:
In the first iteration of the code, I used a simple `stringify()` comparison, and that resulted in globals never matching. One of the problems was the logger, because sometimes it would be stringified as an object, sometimes it would not be in the output, and sometimes it would be detected as a circular object, and so would change the output yet again. The conditions on L126-L127 were the ones that resulted in the correct comparison. For reference, L134 is never actually reached in our specs, which is an indication that the code above it does what it's supposed to do :)

Reply:
Ok... So the thing with the logger indicates a bug and this is a workaround.. I would even maybe argue, we shouldn't provide the logger in the globals ever - the right usage is `hyper.logger.log...`, and these options are fed into modules constructors, where logging shouldn't be a thing, if there's an error encountered, exception should be thrown...

I'm ok with workaround too, but we should indicate that it is a workaround in a comment. Or, get rid of the logger alltogether and get rid of the workaround.

Reply:
The problem is that the logger is needed during module loading for [sys/table](https://github.com/wikimedia/restbase/blob/1f021307c8f6b2b5f69cf1dfd621d385c528f3b9/sys/table.js#L10) and its dependencies, which is why we add it to the options.